Dockerfile

```diff
@@ -0,0 +1,28 @@
+FROM rust:1.54.0-buster as planner
+WORKDIR /app
+RUN curl https://getsubstrate.io -sSf | bash -s \
+    && apt-get clean && rm -rf /var/lib/apt/lists/*
+COPY . .
+RUN cargo build --release \
+    && mv target/release/node-template /usr/local/bin/supra-dht \
+    && mv /usr/local/cargo/bin/sub* /usr/local/bin/ \
+    && rm -rf target \
+    && rm -rf /root/.cargo/registry/ \
+    && rm -rf /tmp && rm -rf /usr/local/rustup && rm -rf /usr/local/cargo
+
+FROM rust:1.54.0-buster as runtime
```

Comment:
Do we need a rust runtime after the binary has been built?

Reply:
No, in fact it does work on `debian:buster-slim` as well. 
But we really want it to be running on even slimmer `alpine` and I could not make it compile for `x86_64-unknown-linux-musl` target. And we perhaps need a `cacher` stage as well which uses `cargo-chef`; but for now I just let it be the same distro for all the stages.
But, changing it to `debian:buster-slim` would still be sensible for now.